src/engine_core/src/input/InputManager.hpp

```diff
@@ -5,18 +5,67 @@
 */
 
 #pragma once
-#include "KeyData.hpp"
+#include "definitions/KeyData.hpp"
+#include "definitions/MouseButton.hpp"
+#include "definitions/MouseData.hpp"
+#include <glm/vec2.hpp>
 #include <map>
 
 class InputManager
 {
   public:
-    static bool IsKeyDown(KeyCode key);
+    /// <summary>
+    /// Evaluates to true the frame the key is identified as <seealso cref="EKeyState::Pressed"/>
+    /// </summary>
```

Comment:
Try to use code comments based on oxygen as VS XML-based code comments might not work with other editors and when generating the docs with doxygen